src/sentry/ingest/transaction_clusterer/rules.py

```diff
@@ -111,16 +114,33 @@ def _get_rules(project: Project) -> RuleSet:
     return ProjectOptionRuleStore().read(project)
 
 
-def get_sorted_rules(project: Project) -> List[Tuple[ReplacementRule, int]]:
+def get_sorted_rules(
+    project: Project, amount: Optional[int] = None
+) -> List[Tuple[ReplacementRule, int]]:
     """Public interface for fetching rules for a project.
 
     The rules are fetched from project options rather than redis, because
     project options is the more persistent store.
 
-    The rules are ordered by specifity, meaning that rules that go deeper
-    into the URL tree occur first.
+    `amount` is the threshold for the number of rules that the function
+    returns. Returned rules are always the most recently used ones, and they are
+    sorted by depth (deeper in the URL tree go first) and usage (more recently
+    used go first).
     """
-    return ProjectOptionRuleStore().read_sorted(project)
+    rules = ProjectOptionRuleStore().read_sorted(project)
+    # Even if there are fewer rules than the given amount, sort them. Keeping
+    # always the same order allows the rules to disappear for not being used.
+    rules = sorted(rules, key=lambda r: r[1], reverse=True)
```

Comment:
Sorting by latest discovery is the best we can do for now, but after we implement the `last_used` timestamp for rules (see https://github.com/getsentry/team-ingest/issues/14), we should use that timestamp instead. The oldest rule might well be the most frequently used.